File: 0959-regions-cut-by-slashes/0959-regions-cut-by-slashes.cpp
class Solution {
public:
    
    int dfs(vector<vector<int>>&v,int i,int j){
        if(min(i,j)<0 || max(i,j)>=v.size()||v[i][j]!=0)return 0;
        v[i][j]=1;
        return 1+dfs(v,i+1,j)+dfs(v,i,j+1)+dfs(v,i-1,j)+dfs(v,i,j-1);
    }
    
    int regionsBySlashes(vector<string>& grid) {
        int cnt=0;
        vector<vector<int>>vect(grid.size()*3,(vector<int>(grid[0].size()*3,0)));
        for(int i=0;i<grid.size();i++){
            for(int j=0;j<grid[0].size();j++){
                if(grid[i][j]=='/')
                    vect[i*3][j*3+2]=vect[i*3+1][j*3+1]=vect[i*3+2][j*3]=1;
                else if(grid[i][j]=='\\')
                    vect[i*3][j*3]=vect[i*3+1][j*3+1]=vect[i*3+2][j*3+2]=1;
            }
        }
        
        for(int i=0;i<grid.size()*3;i++){
            for(int j=0;j<grid[0].size()*3;j++){
                cnt+=dfs(vect,i,j)?1:0;
            }
        }
        return cnt;
    }
};

File: 0959-regions-cut-by-slashes/0959-regions-cut-by-slashes.java
class Solution {
    
    int dfs(int [][]g,int i,int j){
        if(Math.min(i,j)<0 || Math.max(i,j)>=g.length||g[i][j]!=0)return 0;
        g[i][j]=1;
        return 1+dfs(g,i+1,j)+dfs(g,i,j+1)+dfs(g,i,j-1)+dfs(g,i-1,j);
    }
    
    public int regionsBySlashes(String[] grid) {
        
        int n=grid.length,cnt=0;
        int [][]v=new int[n*3][n*3];
        for(int i=0;i<n;i++){
            for(int j=0;j<n;j++){
                if(grid[i].charAt(j)=='/'){
                    v[i*3][j*3+2]=v[i*3+1][j*3+1]=v[i*3+2][j*3]=1;
                }
                else if(grid[i].charAt(j)=='\\'){
                    v[i*3][j*3]=v[i*3+1][j*3+1]=v[i*3+2][j*3+2]=1;
                }
            }
        }
        for(int i=0;i<n*3;i++){
            for(int j=0;j<n*3;j++){
                cnt+=dfs(v,i,j)>0?1:0;
            }
        }
        return cnt;
    }
}